app/api/ms_api_utils.py

```diff
@@ -23,17 +19,69 @@
 # create instance
 def post_request(request_url, data):
     try:
-
         response = requests.post(
             request_url, json=data, headers={"Accept": "application/json"}
         )
         response.raise_for_status()
-    except requests.ConnectionError as e:
-        return f"{e.response.json()}", HTTPStatus.INTERNAL_SERVER_ERROR
-    except requests.HTTPError as e:
-        return f"{e.response.json()}", e.response.status_code
+        response_message = response.json()
+        response_code = response.status_code
+    except requests.exceptions.ConnectionError as e:
+        response_message = messages.INTERNAL_SERVER_ERROR
+        response_code = json.dumps(HTTPStatus.INTERNAL_SERVER_ERROR)
+        logging.warning(f"{e}")
+        return response_message, response_code
+    except requests.exceptions.HTTPError as e:
+        response_message = e.response.json()
+        response_code = e.response.status_code
+        return response_message, response_code
     except Exception as e:
-        return f"{e.response.json()}", e.response.status_code
+        response_message = messages.INTERNAL_SERVER_ERROR
+        response_code = json.dumps(HTTPStatus.INTERNAL_SERVER_ERROR)
+        logging.warning(f"{e}")
+        return response_message, response_code
     else:
         logging.warning(f"{response}")
-        return f"{response.json()}", response.status_code
+        return response_message, response_code
+
+    
+def http_response_checker(result, request_type):
+    
+    if result[1] == HTTPStatus.OK:
+        return http_ok_status_checker(result, request_type)
+    if result[1] == HTTPStatus.BAD_REQUEST:
+        return http_bad_request_checker(result)
+    
+    # TO DO. CHANGE HTTPStatus.NOT_FOUND to UNAUTHORIZED AFTER PR647 IN MS BACKEND IS APPROVED
+    if request_type == "login":
```

Comment:
Can make these 3 lines a separate function

Reply:
When the PR647 in MS got approved (which will be soon, hopefully), this code will no longer be 3 lines. It'll just be 2 lines like the rest. Can we keep it here for now?

Reply:
Sure